src/foam/core/CompoundException.java

```diff
@@ -32,7 +32,10 @@ public void add(Throwable t) {
 
   public void maybeThrow() {
     if ( exceptions_.size() != 0 ) {
-      throw this;
+      for ( int i = 0 ; i < exceptions_.size() ; i++ ) {
```

Comment:
@hchoura no.
Do this output when AppConfig.MODE is TEST.
after this block 
'throw this' as before.